### ISSUE TITLE
Rename Subgame to Game

### ISSUE THREAD

**Issue by rubenwardy** [minetest/minetest#7053]
Subgame is a very weird term that is odd to beginners. If Minetest is aiming to be an engine, then it should be an engine which runs games. Not things that are less than games.

If this is agreed upon, then all user-facing mentions should be changed to "game" rather than "subgame", and then other sources should be updated (ie: forum sections, website)

- [x] Update engine
- [x] Update MTG - https://github.com/minetest/minetest_game/pull/2081
- [ ] Update website - https://github.com/minetest/minetest.github.io/pull/122
- [x] Rename forums
- [ ] Update wikis (I've done this particularly, needs checking for consistency)
- [ ] Release news post (I have one written)

This issue is unanimously approved by users. Devs:

* 7 in favour: @rubenwardy, @paramat, @sofar, @Ezhh, @nerzhul, @celeron55, @sfan5.
* 1 neutral: @SmallJoker (disagrees, but notices user favour).

**Comment by Ezhh** [comment]
Yes please. We simply cannot call it a game engine if it does not run something we call games.

**Comment by cx384** [comment]
The expression "game" seems a little bit too general to me, but I haven't got an idea for another option.

**Comment by 4w** [comment]
Yes, so-called “subgames” should be referenced as “Games” in code and configuration. It just makes more sense.

**Comment by NathanSalapat** [comment]
Subgame does sound like something that is part of another game, which in this case they aren't. Renaming to game sounds good to me.

**Comment by paramat** [comment]
Excellent, i was actually about to suggest the same.
In almost all of the code and in lua_api.txt 'game' is already correctly used, MTG is not 'Minetest Subgame' and we have a 'games' folder etc.
'Subgame' has crept in in certain places like on the forum, driven by a silly concern that MT is seen as a 'game' so anything that runs on it should be called a 'subgame'. In fact 'subgame' is likely to cause more confusion.
Every time the word 'subgame' is used it irritates me and i am conciously stopping myself using it.

**Comment by Fixer-007** [comment]
Minetest Game is Game, subgame is game that uses Minetest Game as its base, pretty much, I support renaming it to Game.

**Comment by paramat** [comment]
We do have a 'subgame.cpp' file, maybe rename to 'games.cpp'?

**Comment by MuhdNurHidayat** [comment]
I have never thought why it is subgame when it should be game from the beginning. I've been thinking it's some sort of technical name thingy.

I would agree if subgame is renamed to game as it's easier to understand. To be honest, the word 'subgame' sounds like they're mod (as in they're 'something as a part of a game'), which is not really true because the thing we call subgame is by itself a complete game to be run on Minetest engine.

**Comment by C1ffisme** [comment]
Subgame sounds a lot like "minigame", which doesn't make sense considering a subgame is a game built off an API while a minigame is a small part of a game that deviates from regular gameplay.

Heck, the term "subgame" suggests that it's part of a much larger game, but it's not really a part of anything, it's just a bunch of function calls that set up the rules of a game built on top of an engine with no default gameplay whatsoever.

TL;DR : :+1: for using less confusing terminology.

**Comment by nerzhul** [comment]
okay for me. subgame means there is a root game, and in fact root game is the subgame. Then go

**Comment by celeron55** [comment]
Fine for me too.

**Comment by sfan5** [comment]
Fine by me.

**Comment by appgurueu** [comment]
I agree totally.

**Comment by paramat** [comment]
Please also try to write 'game' whereever you are discussing MT, as common usage in text tends to spread.

**Comment by rubenwardy** [comment]
I believe we have a mandate. Now a PR needs to made to the engine to rename any user facing text from subgame to game. Internal references to "Subgame" is fine short term, imo

**Comment by paramat** [comment]
Doing a search with 'subgame', we have 'subgame.cpp' and 'subgame' used in cpp code, that is needed to distinguish from 'game' and is fine.

There is usage in builtin, in translation files and in docs that should be changed.

**Comment by 4w** [comment]
> There is usage in builtin, in translation files and in docs that should be changed, it's not much to do.

Don't forget the forums.

![image](https://user-images.githubusercontent.com/30605057/37138739-86db2a20-22ab-11e8-8c4b-05ef29dffcde.png)

**Comment by 4w** [comment]
![image](https://user-images.githubusercontent.com/30605057/37138770-a95e71e2-22ab-11e8-93d4-e10af69e5020.png)

WTF Github. No, I didn't. I'm pretty sure I'm not even allowed to ... 😃

**Comment by rubenwardy** [comment]
Now implemented :tada: 

https://forum.minetest.net/viewtopic.php?f=18&t=19808

**Comment by paramat** [comment]
All ticked?

**Comment by 4w** [comment]
For wiki the built-in search can be used to detect where “subgame” is used.

https://wiki.minetest.net/index.php?search=subgame&title=Special%3ASearch&fulltext=1  
http://dev.minetest.net/index.php?search=subgame&title=Special%3ASearch&fulltext=1
